src/test/java/spoon/test/main/MainTest.java

```diff
@@ -439,6 +435,35 @@ public void scan(CtRole role, CtElement element) {
 		});
 	}
 
+	@Test
+	public void testElementToPathToElementEquivalency() {
+
+		List<CtElement> list = new LinkedList<>();
+		list.add(rootPackage);
+
+		rootPackage.accept(new CtScanner() {
+			@Override
+			public void scan(CtRole role, CtElement element) {
+				if (element != null) {
+					CtPath path = element.getPath();
+					String pathStr = path.toString();
+					try {
+						CtPath pathRead = new CtPathStringBuilder().fromString(pathStr);
+						Collection<CtElement> returnedElements = pathRead.evaluateOn(list);
+						//contract: CtUniqueRolePathElement.evaluateOn() returns a unique elements if provided only a list of one inputs
+						assertEquals(returnedElements.size(), 1);
+						CtElement actualElement = (CtElement) returnedElements.toArray()[0];
+						//contract: Element -> Path -> String -> Path -> Element leads to the original element
+						assertEquals(element, actualElement);
```

Comment:
assertSame?

Reply:
Yes, that sounds better. I'll fix it ASAP.